packages/common/src/llm/schemas.ts

```diff
@@ -0,0 +1,99 @@
+import { z } from '@botpress/sdk'
+
+const ToolCallSchema = z.object({
+  id: z.string(),
+  type: z.enum(['function']),
+  function: z.object({
+    name: z.string(),
+    arguments: z.string(),
+  }),
+})
+
+export type ToolCall = z.infer<typeof ToolCallSchema>
+
+const ToolChoiceSchema = z.object({
+  // TODO: remove empty value from enum once Studio issue is fixed
+  type: z.enum(['auto', 'specific', 'any', 'none', '']).optional(), // note: Claude doesn't support "none" but we can simply strip out the tools when `type` is "none"
+  functionName: z.string().optional().describe('Required if `type` is "specific"'),
+})
+
+const MessageSchema = z.object({
+  role: z.enum(['user', 'assistant']),
+  type: z.enum(['text', 'tool_calls', 'tool_result', 'multipart']),
+  toolCalls: z.array(ToolCallSchema).optional().describe('Required if `type` is "tool_calls"'),
+  toolResultCallId: z.string().optional().describe('Required if `type` is "tool_result"'), // note: not supported by Gemini
+  content: z
+    .string()
+    // TODO: union types are not supported yet by the Studio, comment this out when testing via an action card in the Studio
+    .or(
+      z.array(
+        z.object({
+          type: z.enum(['text', 'image']),
+          mimeType: z.string(),
+          text: z.string().optional().describe('Required if part type is "text" '),
+          url: z.string().optional().describe('Required if part type is "image"'),
```

Comment:
**nitpick:** I think I'd prefer `imageUrl` as we use in a lot of places, but I don't have a strong opinion

Reply:
I chose `url` as we need this to support audio and video as well (will be worked on later) 😄